File: Programmers/Lv.1/최대공약수와최소공배수.py
import math

def solution(n, m):
    return [math.gcd(n, m), n * m / math.gcd(n, m)]

# def solution(a, b):
#     c, d = max(a, b), min(a, b)
#     t = 1

#     while t > 0:
#         t = c % d
#         c, d = d, t
#     answer = [c, int(a * b / c)]

#     return answer

File: Programmers/Lv.1.Kotlin/최대공약수와최소공배수.kt
import kotlin.math.*

class Solution {
    fun solution(n: Int, m: Int): IntArray {
        return intArrayOf(gcm(n, m), lcm(n, m))
    }

    fun gcm(a: Int, b: Int): Int {
        var maximum = max(a, b)
        var minimum = min(a, b)

        if (minimum == 0) {
            return maximum
        } else {
            return gcm(minimum, maximum % minimum)
        }
    }

    fun lcm(a: Int, b: Int): Int = (a * b) / gcm(a, b)

    // fun solution(n: Int, m: Int): IntArray {
    //     val gcd = gcd(n, m)

    //     return intArrayOf(gcd, n * m / gcd)
    // }

    // tailrec fun gcd(a: Int, b: Int): Int = if (b == 0) a else gcd(b, a % b)
}